packages/react/ds/src/table/table.stories.tsx

```diff
@@ -8,40 +13,272 @@ import { TableHeader } from './table-header.js';
+      },
+      {
+        id: 3,
+        county: 'Sustainability Initiative',
+        description: 'Carbon neutrality program',
+        total: '€2,250,000',
+        status: { text: 'Behind Schedule', type: TagType.error },
+      },
+    ],
+  },
+  render: ({ captionText, headers, rows }) => (
+    <Table>
+      <Caption>{captionText}</Caption>
+      <TableHead>
+        <TableRow>
+          {headers?.map((heading, index) => (
+            <TableHeader key={`header-${index}`}>{heading}</TableHeader>
+          ))}
+        </TableRow>
+      </TableHead>
+      <TableBody>
+        {rows?.map((row) => (
+          <TableRow key={`row-${row.id}`}>
+            <TableData>
+              <Checkbox
+                ariaLabel={`checkbox-id-${row.id}`}
+                checkboxId={`project-select-${row.id}`}
+                value={row.id.toString()}
+                dataElement={`project-checkbox-${row.id}`}
+              />
+            </TableData>
+            <TableData>{row.county}</TableData>
+            <TableData>
+              <Link href="#">View Profile</Link>
+            </TableData>
+            <TableData>{row.total}</TableData>
+            <TableData>
+              <Tag text={row.status.text} type={row.status.type} />
+            </TableData>
+            <TableData>
+              <div className="gi-flex">
+                <IconButton
+                  icon={{ icon: 'edit' }}
+                  onClick={() => alert(`Edit project ${row.id}`)}
+                  variant="flat"
+                  size="large"
+                />
+                <IconButton
+                  icon={{ icon: 'delete' }}
+                  onClick={() => alert(`Delete project ${row.id}`)}
+                  variant="flat"
+                  size="large"
+                />
+              </div>
+            </TableData>
           </TableRow>
-        </TableFoot>
-      </>
-    ),
+        ))}
+      </TableBody>
+      <TableFoot>
+        <TableRow>
+          <TableData colSpan={3} className="gi-font-bold gi-text-right gi-py-4">
+            Total Budget:
+          </TableData>
+          <TableData colSpan={3} className="gi-font-bold gi-text-left gi-py-4">
+            €4,500,000
+          </TableData>
+        </TableRow>
+      </TableFoot>
+    </Table>
+  ),
+};
+
+export const EmptyState: Story = {
+  args: {
+    captionText: 'No Data Available',
+    headers: ['ID', 'Name', 'Status'],
+    rows: [],
   },
+  render: ({ captionText, headers = [] }) => (
+    <Table>
+      <Caption>{captionText}</Caption>
+      <TableHead>
+        <TableRow>
+          {headers?.map((heading, index) => (
+            <TableHeader key={`header-${index}`}>{heading}</TableHeader>
+          ))}
+        </TableRow>
+      </TableHead>
+      <TableBody>
+        <TableRow>
+          <TableData
+            colSpan={headers.length}
+            className="!gi-text-center !gi-border-0 !gi-py-8 !gi-px-4"
```

Comment:
do we need `!` in classes?

Reply:
Please avoid using `!important` classes.
Check #352 PR

Can you have an `EmptyState` component that goes into `TableData` slot for that case?